### PR TITLE
feat(INVT-CPC-734): Update product inside of inventory

### PR DESCRIPTION
**Pull request by AlexeiTimbro**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/CPC-734

What is the ticket about and why are we doing this change.

This pull request is about the backend implementation of the Update Product In Inventory feature of the inventory Service.

## Changes
- Fully working backend methods UpdateProductInInventory
- ControllerUnitTest
- ControllerIntegrationTest
- ProductRepositoryIntegrationTest
- ServiceUnitTest

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java [hunk 1]
```diff
@@ -61,6 +61,16 @@ public Mono<InventoryResponseDTO> updateInventory(InventoryRequestDTO model, Str
...
                 .retrieve().bodyToMono(InventoryResponseDTO.class);
     }
+
+    public Mono<ProductResponseDTO> updateProductInInventory(final ProductRequestDTO model, final String inventoryId, final String productId){
+        return webClient
+                .put()
+                .uri(inventoryServiceUrl + "/{inventoryId}/products/{productId}", inventoryId, productId)
+                .body(Mono.just(model),ProductRequestDTO.class)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve().bodyToMono(ProductResponseDTO.class);
```
Comment (nic5694) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java:71]: you should be checking the fields in the response model to make sure they are equivalent to what is expected

File: api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java [hunk 2]
```diff
@@ -2086,8 +2084,47 @@ void deleteAllProductInventory_shouldSucceed() {
...
 
+        // Perform the PUT request
+        client.put()
+                .uri("/api/gateway/inventory/{inventoryId}/products/{productId}", "sampleInventoryId", "sampleProductId")
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue(requestDTO)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody()
```
Comment (nic5694) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java:2115]: in the expectBody put the corresponding response model

File: inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerIntegrationTest.java [hunk 3]
```diff
@@ -453,6 +472,131 @@ void deleteAllInventories_ShouldDeleteAllInventoriesAndAssociatedProducts() {
...
+                .expectBody(ProductResponseDTO.class)
+                .value(productResponseDTO -> {
+                    assertNotNull(productResponseDTO);
+                    assertEquals(productId, productResponseDTO.getProductId());
+                    assertEquals(productRequestDTO.getProductName(), productResponseDTO.getProductName());
+                    assertEquals(productRequestDTO.getProductDescription(), productResponseDTO.getProductDescription());
+                    assertEquals(productRequestDTO.getProductPrice(), productResponseDTO.getProductPrice());
+                    assertEquals(productRequestDTO.getProductQuantity(), productResponseDTO.getProductQuantity());
+                });
+    }
```
Comment (nic5694) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerIntegrationTest.java:597]: remove ALL this commented code their should be no commented code in production code

File: inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerUnitTest.java [hunk 4]
```diff
@@ -343,6 +343,160 @@ void deleteProductInventory_ValidInventoryId_ShouldCallServiceDelete() {
...
+                .jsonPath("$.message", "Invalid input");
+
+        verify(productInventoryService, times(1))
+                .updateProductInInventory(any(), eq(inventoryId), eq(productId));
+
+        verify(productInventoryService, times(1)).deleteAllProductInventory(inventoryId);
+
+    }
+    
+ */
```
Comment (nic5694) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerUnitTest.java:494]: remove the commented code, no commented code should be in the production code

**Review by nic5694 — CHANGES_REQUESTED** [review]
Please review the comments i left you before merging to main.

**Review by nic5694 — APPROVED** [review]
perfect, thank you for applying the changes

**Review by Iantomasi — APPROVED** [review]
Everything looks good, passes the vibe check.

**Pull request merged**